### PR TITLE
Add 16 bit support to kernel operator TRANSPOSE_CONVOLUTION

### PR DESCRIPTION
**Pull request by psunn**
This PR is one of steps to extend TFLite to support symmetric 16-bit activations.

In this PR we introduce implementation and tests for TRANSPOSE_CONV kernel reference function.
The specification of this operator:

TRANSPOSE_CONVOLUTION
  Input 0:
    data_type  : int16
    range      : [-32768, 32767]
    granularity: per-tensor, zero_point=0
  Input 1 (Weight):
    data_type  : int8
    range      : [-127, 127]
    granularity: per-axis (dim = 0), zero_point=0
  Output 0:
    data_type  : int16
    range      : [-32768,32767]
    granularity: per-tensor, zero_point=0

### CODE REVIEW

**Comment by psunn** [comment]
This PR is relative to [TransposeConv with Bias](https://github.com/tensorflow/tensorflow/pull/34903)
I can update this PR if the other merged, or update the other PR if this merged first.
We can discuss how to resolve it.

**Comment by psunn** [comment]
Hi @suharshs @jdduke
Could you please review this?
As discussed, this one of the INT16 reference kernel reference function.
Thanks!

**Review by suharshs — DISMISSED** [review]

File: tensorflow/lite/kernels/internal/compatibility.h [hunk 1]
```diff
@@ -83,6 +83,8 @@ using int16 = std::int16_t;
 using uint16 = std::uint16_t;
 using int32 = std::int32_t;
 using uint32 = std::uint32_t;
+using int64 = std::int64_t;
+using uint64 = std::uint64_t;
```
Comment (rthadur) [inline tensorflow/lite/kernels/internal/compatibility.h]: `In file included from ./knowledge/cerebra/sense/text_classifier/training/augmenter.h:9:
./knowledge/cerebra/sense/text_classifier/utils/random-source.h:38:27: error: reference to 'int64' is ambiguous
  explicit MtRandomSource(int64 seed);
                          ^
./third_party/tensorflow/lite/kernels/internal/compatibility.h:86:7: note: candidate found by name lookup is 'int64'
using int64 = std::int64_t;
      ^
third_party/absl/base/integral_types.h:75:29: note: candidate found by name lookup is 'int64'
typedef long long           int64;
                            ^
1 error generated.

In file included from knowledge/cerebra/sense/text_classifier/training/augmenter.cc:1:
In file included from ./knowledge/cerebra/sense/text_classifier/training/augmenter.h:9:
./knowledge/cerebra/sense/text_classifier/utils/random-source.h:38:27: error: reference to 'int64' is ambiguous
  explicit MtRandomSource(int64 seed);
                          ^
./third_party/tensorflow/lite/kernels/internal/compatibility.h:86:7: note: candidate found by name lookup is 'int64'
using int64 = std::int64_t;
      ^
third_party/absl/base/integral_types.h:75:29: note: candidate found by name lookup is 'int64'
typedef long long           int64;
                            ^`

Comment (rthadur) [reply tensorflow/lite/kernels/internal/compatibility.h]: please check above error ,this is failing internally.

**Review by rthadur — CHANGES_REQUESTED** [review]

File: tensorflow/lite/kernels/internal/compatibility.h (hunk 1 above)
Comment (psunn) [reply tensorflow/lite/kernels/internal/compatibility.h]: removed ambiguous reference. please have a look.

**Review by suharshs — DISMISSED** [review]

**Comment by marksunpeng** [comment]
I have a look on `Windows Bazel` build and ` Windows Bazel GPU` build:
none of the logs seems failing relative to this PR
https://source.cloud.google.com/results/invocations/ea48732b-6791-400f-9343-274a6bdce49f/log
https://source.cloud.google.com/results/invocations/1f43a2e1-15b2-4cbf-b0c8-a098297e2eed/log

@suharshs @rthadur can you help to have a look?
Shall I rebase this PR?

**Comment by psunn** [comment]
I have a look on Windows Bazel build and Windows Bazel GPU build:
none of the logs seems failing relative to this PR
https://source.cloud.google.com/results/invocations/ea48732b-6791-400f-9343-274a6bdce49f/log
https://source.cloud.google.com/results/invocations/1f43a2e1-15b2-4cbf-b0c8-a098297e2eed/log

@suharshs @rthadur can you help to have a look?
Shall I rebase this PR?

**Comment by suharshs** [comment]
I am still  making some internal tests pass with this change. Should be in
soon.

**Comment by rthadur** [comment]
@suharshs gentle ping for new status ?

**Comment by jdduke** [comment]
Can we rebase? I think there are some stale errors for this one.

**Review by jdduke — APPROVED** [review]

**Review by rthadur — DISMISSED** [review]

**Comment by rthadur** [comment]
Here is the internal error , iam not sure if it is related , can you please verify 

`third_party/tensorflow/lite/kernels/test_util.cc:330
Expected equality of these values:
  CountPartitionsDelegatedTo(interpreter_.get(), TestNnApiDelegate())
    Which is: 0
  1
Expecting operation to be accelerated but cannot find a partition associated to the NNAPI delegate`

**Comment by jdduke** [comment]
I think we just need to blacklist the new int16 tests @ /tensorflow/lite/delegates/nnapi/acceleration_test_list.cc.

**Comment by psunn** [comment]
I'm wondering, is add a version to int16 TRANSPOSE_CONV gonna be helpful?

**Review by jdduke — DISMISSED** [review]

**Comment by gbaned** [comment]
@psunn Can you please resolve conflicts? Thanks!

**Comment by marksunpeng** [comment]
@gbaned thanks for the reminder. done

@jdduke @suharshs  Can you please have a review? thanks!

**Comment by psunn** [comment]
@gbaned thanks for reminding me, done
@jdduke @suharshs can you please review it? thanks!

**Comment by psunn** [comment]
@jdduke sorry to bother you, can you have a review and merge if it's ok?

**Comment by jdduke** [comment]
I think @suharshs is working to merge this manually.

**Review by jdduke — APPROVED** [review]

**Review by rthadur — APPROVED** [review]

**Pull request merged**